src/linear_algebra/matrix/symmetric.lean

```diff
@@ -0,0 +1,134 @@
+/-
+Copyright (c) 2021 Lu-Ming Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Lu-Ming Zhang
+-/
+import data.matrix.block
+
+/-!
+# Symmetric matrices
+
+This file contains the definition and basic results about symmetric matrices.
+
+## Main definition
+
+ * `matrix.is_symm `: a matrix `A : matrix n n α` is "symmetric" if `Aᵀ = A`.
+
+## Tags
+
+symm, symmetric, matrix
+-/
+
+variables {α β n m R : Type*}
+
+namespace matrix
+
+open_locale matrix
+
+/-- A matrix `A : matrix n n α` is "symmetric" if `Aᵀ = A`. -/
+def is_symm (A : matrix n n α) : Prop := Aᵀ = A
+
+lemma is_symm.eq {A : matrix n n α} (h : A.is_symm) : Aᵀ = A := h
+
+lemma is_symm.ext_iff {A : matrix n n α} : A.is_symm ↔ ∀ i j, Aᵀ i j = A i j :=
+by rw [is_symm, matrix.ext_iff]
+
+lemma is_symm.ext_iff' {A : matrix n n α} : A.is_symm ↔ ∀ i j, A j i = A i j :=
+is_symm.ext_iff
+
+lemma is_symm.apply {A : matrix n n α} (h : A.is_symm) (i j : n) : Aᵀ i j = A i j :=
+is_symm.ext_iff.1 h i j
+
+lemma is_symm.apply' {A : matrix n n α} (h : A.is_symm) (i j : n) : A j i = A i j :=
+is_symm.apply h i j
+
+lemma is_symm.ext {A : matrix n n α} : (∀ i j, Aᵀ i j = A i j) → A.is_symm :=
+is_symm.ext_iff.2
+
+lemma is_symm.ext' {A : matrix n n α} : (∀ i j, A j i = A i j) → A.is_symm :=
+is_symm.ext
+
+/-- `A ⬝ Aᵀ` is symmertric. -/
+lemma is_symm_mul_transpose_self [fintype n] [comm_semiring α] (A : matrix n n α) :
+  (A ⬝ Aᵀ).is_symm :=
+transpose_mul _ _
+
+/-- `Aᵀ ⬝ A` is symmertric. -/
+lemma is_symm_transpose_mul_self [fintype n] [comm_semiring α] (A : matrix n n α) :
+  (A ⬝ Aᵀ).is_symm :=
+transpose_mul _ _
+
+/-- `A + Aᵀ` is symmertric. -/
```

Comment:
```suggestion
/-- `A + Aᵀ` is symmetric. -/
```
This typo is repeated a lot.

Arguably you could just remove all these comments - they look exactly the same as the theorem statement, it's like writing
```python
i = i + 1  # add one to i
```
which is usually frowned upon.

Reply:
That is a "copy-paste" issue. I will delete them.